### DISCUSSION TITLE
How to disable Folders as Search Result

### DISCUSSION THREAD

**Discussion by 1nY0urF4c3** [microsoft-search/pnp-modern-search#2124] [Q&A] [answered]
Hi all, 

I'm sure it's possible but I couldn't find any content on this topic. Please excuse this "beginner" question but how can I exclude folders from results in my results webpart? I am working with V4 of PNP Search and I want to show results of a single library without folders. 

Many greetings

**Comment by FranckyC** [comment, marked as answer]
In the query template you can use this condition `-ContentTypeId:0x0120* IsDocument:1 Path:"https://<your_library_path>/*"`

**Reply by 1nY0urF4c3** [reply]
Thank you very much and sorry for the circumstances. TY u saved me..

**Reply by wobba** [reply]
This will also exclude the OneNote entity or DocumentSet as those are of type folder technically as well. Just so you are aware of any side effects.

**Reply by 1nY0urF4c3** [reply]
Thank you very much for the additional informations. In this special case ist totally fine for me.

**Reply by NumrexRafia** [reply]
> In the query template you can use this condition `-ContentTypeId:0x0120* IsDocument:1 Path:"https://<your_library_path>/*"`

Hi, @FranckyC  @wobba  I am stuck in a scenario kindly help... well I want my search result web part to show only the available sites on my tenant (only the links to the site) and no other content(mean no document libraries/lists/pages etc) I tried to add in QueryTemplate the ContentTypeId: xxx-xx-xx-xx now it's showing me the available page as well but I want only the sites.  in another scenario, I want to display only the list names but it's showing the list items as well kindly help me with this.

**Reply by wobba** [reply]
`contentclass:sts_site OR contentclass:sts_web` will return only sites
